features/prog.go

```diff
@@ -0,0 +1,157 @@
+package features
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func init() {
+	pc.progTypes = make(map[ebpf.ProgramType]error)
+}
+
+var (
+	pc progCache
+)
+
+type progCache struct {
+	sync.Mutex
+	progTypes map[ebpf.ProgramType]error
+}
+
+func createProgLoadAttr(pt ebpf.ProgramType) (*internal.BPFProgLoadAttr, error) {
+	var expectedAttachType ebpf.AttachType
+
+	insns := asm.Instructions{
+		asm.LoadImm(asm.R0, 0, asm.DWord),
+		asm.Return(),
+	}
+
+	buf := bytes.NewBuffer(make([]byte, 0, len(insns)*asm.InstructionSize))
+	err := insns.Marshal(buf, internal.NativeEndian)
+	if err != nil {
```

Comment:
Could be
```suggestion
	if err := insns.Marshal(buf, internal.NativeEndian); err != nil {
```
Here and below in the same function.

Reply:
I don't think I can apply this below with `v, err := internal.KernelVersion()` because then variable `v` is only available in the if scope I believe, so I would have to use an `else` which I'd like to avoid. I'll use your suggestion for the first occurrence though :)

Reply:
Correct I read too fast for the second one :+1: